### PR TITLE
Use regex to match badwords

### LINKED ISSUE #20: Change word list to be coma delimited instead of space delimited
The word list is currently space delimited. This is not ideal because in some cases "bad word" should be filtered but "bad" and "word" are not offensive at all in a different context when not next to each other. 

This should be coma delimited so users can specify to filter on "bad word" but users can post "bad" or "word".  

Note that the current list already includes terms that have spaces in them which would cause unexpected words to be blocked.

### LINKED ISSUE #22: Profanity filter should ignore punctuation when looking for profanity
Currently the profanity filter only looks for words that are an exact match on the "Bad words list" and nothing with punctuation is filtered out. So if I type "expletive" it will get filtered by the pluign but, "expletive!" or "expletive." with the punctuation, the filter has no affect. 

Because it very common, we need to be able to filter words that are followed by punctuation. Maybe we could use some type of regex match instead?

### PR DESCRIPTION
**Pull request by nathanaelhoun**
#### Summary

Use regex to match badwords.

The regex is build on configuration change and then use for filter each post.

- [x] The plugin can detect multi-word badwords (**Breaking changes : the word list is now comma-separated**)
- [x] The plugin now detect badwords even with punctuation
- [x] Default bad word list is updated to add more multi-word bad words and use regex
- [x] `README.md` updated

image

#### Ticket Link

Closes #20 and closes #22

### CODE REVIEW

**Review by hanzei — APPROVED** [review]
LGMT :+1:

File: server/plugin.go [hunk 1]
```diff
@@ -23,29 +24,30 @@ type Plugin struct {
...
-	words := strings.Split(message, " ")
-	for i, word := range words {
-		if p.WordIsBad(word) {
-			if configuration.RejectPosts {
-				return nil, "Profane word not allowed: " + word
-			}
-			words[i] = strings.Repeat(configuration.CensorCharacter, len(word))
+	postWithoutAccents := removeAccents(post.Message)
+
+	if p.badWordsRegex.MatchString(postWithoutAccents) {
```
Comment (levb) [inline server/plugin.go:33]: nit: 1/5 `if !match { return post, nil }` - for better typography

**Review by levb — APPROVED** [review]
LGTM, with a nit.

**Comment by levb** [comment]
@nathanaelhoun thank you for this meaningful improvement. Do you mind fixing my nit comment, if you agree with it? Otherwise, LGTM. @DHaussermann will you have the cycles to test this, or should we consider the new unit tests sufficient?

**Comment by DHaussermann** [comment]
Yes, I can take a quick look once changes are completed.

File: server/plugin.go (hunk 1 above)
Comment (nathanaelhoun) [reply server/plugin.go:33]: Agree

**Comment by nathanaelhoun** [comment]
@levb Thanks! Agree with you, so I applied it.

@DHaussermann You can test this :-)

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Regex match solves both issues as expected
- Can now filter words that contain a space like "bad word" with no impact on the individual word
- Tested basic ways of circumventing filter:
  - add ! and other punctuation
  - add dash or other special char at the end
  - encase the word in a `comment block` 
- Users will no longer side-step the filter naturally by punctuating words
- No issues found. 

LGTM!

Thanks @nathanaelhoun Much appreciated. You've added a huge amount of value by making this change!

**Pull request merged**